### PR TITLE
A couple of bugfixes 

### PR DESCRIPTION
**Pull request by felixmaximilian**
fixed wrong indexing into dict; every picture is read into 3 channels even if input is grayscale; loaded jpeg is read as rb buffer.

### CODE REVIEW

**Comment by felixmaximilian** [comment]
I signed it!

**Comment by felixmaximilian** [comment]
One fix of this PR seems to aid the following issue: #2304 .

**Comment by felixmaximilian** [comment]
[](url)

File: tensorflow/examples/image_retraining/retrain.py [hunk 1]
```diff
@@ -616,7 +616,7 @@ def add_input_distortions(flip_left_right, random_crop, random_scale,
   """
 
   jpeg_data = tf.placeholder(tf.string, name='DistortJPGInput')
-  decoded_image = tf.image.decode_jpeg(jpeg_data)
+  decoded_image = tf.image.decode_jpeg(jpeg_data,channels=MODEL_INPUT_DEPTH)
```
Comment (mrry) [inline tensorflow/examples/image_retraining/retrain.py]: Style nit: add a space between the `,` and `channels`.

**Comment by felixmaximilian** [comment]
Please, can someone explain why the travis ci builds aren't running?

**Comment by felixmaximilian** [comment]
Please test this, Jenkins.

**Comment by vrv** [comment]
Seems fine but are there any tests that should have caused this to fail?  cc @petewarden

**Comment by petewarden** [comment]
Good catch! It would have been good to catch this with a test, but the training apparently still produced good results even with black and white images, so it wasn't obvious.

Jenkins, test this please.

**Pull request merged**